Gordon360/Services/MembershipRequestService.cs

```diff
@@ -81,7 +81,7 @@ public async Task<RequestView> ApproveAsync(int requestID)
                 throw new BadInputException() { ExceptionMessage = "The request has already been approved."};
             }
 
-            MembershipUploadViewModel newMembership = (MembershipUploadViewModel) request;
+            MembershipUploadViewModel newMembership = MembershipUploadViewModel.FromRequest(request, _accountService.GetAccountByID(request.ID_NUM.ToString()).ADUserName);
             newMembership.Username = _accountService.GetAccountByID(request.ID_NUM.ToString()).ADUserName;
```

Comment:
I agree that it's cleaner to add the username during the conversion, rather than hoping that it's added afterwards.

I do think it's easier to read if getting the username is on a separate line though:
```suggestion
            var username = _accountService.GetAccountByID(request.ID_NUM.ToString()).ADUserName;
            MembershipUploadViewModel newMembership = MembershipUploadViewModel.FromRequest(request, username);
```